### PR TITLE
[SYCL] Add ext_oneapi_unified_runtime SYCL backend

### PR DESCRIPTION
**Pull request by smaslov-intel**
This PR adds new backend to SYCL: ext_oneapi_unified_runtime
E2E test: https://github.com/intel/llvm-test-suite/pull/1450

Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

**Review by againull — APPROVED** [review]

**Comment by smaslov-intel** [comment]
@againull : please re-review, the only essential changes are https://github.com/intel/llvm/pull/7816/commits/371f23af6275a5706bf6edc649e9809a5fd6147a and https://github.com/intel/llvm/pull/7816/commits/0374b705fa301399aa7e4a21c966e86cf6894933

**Review by againull — APPROVED** [review]
New changes look good to me.

File: sycl/plugins/unified_runtime/pi2ur.hpp [hunk 1]
```diff
@@ -15,13 +15,7 @@
...
 // Map of UR error codes to PI error codes
 static pi_result ur2piResult(zer_result_t urResult) {
-
-  // TODO: replace "global lifetime" objects with a non-trivial d'tor with
-  // either pointers to such objects (which would be allocated and dealocated
-  // during init and teardown) or objects with trivial d'tor.
-  // E.g. for this case we could have an std::array with sorted values.
-  //
-  static std::unordered_map<zer_result_t, pi_result> ErrorMapping = {
+  std::unordered_map<zer_result_t, pi_result> ErrorMapping = {
```
Comment (againull) [inline sycl/plugins/unified_runtime/pi2ur.hpp:18]: Is this change going to add performance overhead?

Comment (smaslov-intel) [reply sycl/plugins/unified_runtime/pi2ur.hpp:18]: It certainly will, but 1) it should be negligible and 2) it is temporary until we switch to UR.
There is already a map lookup overhead before this change, not this map is re-constructed too.
I'd let the testing see if it causes any real perf issues

Comment (againull) [reply sycl/plugins/unified_runtime/pi2ur.hpp:18]: I see, thanks.

**Review by againull — APPROVED** [review]

**Pull request merged**